In [1]:
from iterative_methods import *
from multi_variate_newton import *
import numpy as np

Error after 0 iteration 0.8558975105001576
Error after 1 iteration 0.5434872554427782
Error after 2 iteration 0.9359007498430079
Error after 3 iteration 0.5690061794192101
Error after 4 iteration 0.5356166277823504
Error after 5 iteration 0.20744843569199925
Error after 6 iteration 0.05287690803517455
Error after 7 iteration 0.0006332361575052114
Error after 8 iteration 5.388830773961401e-06
Error after 9 iteration 2.2154840111600825e-05
Error after 10 iteration 0.0022149473667646743
Error after 11 iteration 0.0
Error after 12 iteration 0.0
Error after 13 iteration 0.0
Error after 14 iteration 0.0
Error after 15 iteration 0.0
Error after 16 iteration 0.0
Error after 17 iteration 0.0
Error after 18 iteration 0.0
Error after 19 iteration 0.0
Error after 20 iteration 0.0
Error after 21 iteration 0.0
Error after 22 iteration 0.0
Error after 23 iteration 0.0
Error after 24 iteration 0.0
Error after 25 iteration 0.0
Error after 26 iteration 0.0
Error after 27 iteration 0.0
Error after 28 ite

/Users/love/PycharmProjects/pythonProject/Exercise-numan/multi_variate_newton.py:73: RuntimeWarning: invalid value encountered in scalar divide
  print(f"Error after {i} iteration", ekp / ek)


In [2]:
#Ex1

def f1(**kwargs):
    x1 = kwargs["x1"]
    x2 = kwargs["x2"]
    x3 = kwargs["x3"]
    return x1 **2 -2*x1 + x2 **2 - x3 + 1

def f2(**kwargs):
    x1 = kwargs["x1"]
    x2 = kwargs["x2"]
    x3 = kwargs["x3"]
    return x1 *x2 **2 - x1 -3*x2 + x2*x3 +2

def f3(**kwargs):
    x1 = kwargs["x1"]
    x2 = kwargs["x2"]
    x3 = kwargs["x3"]
    return x1*x3**2 - 3 * x3 + x2*x3**2 + x1*x2


fs = [f1, f2, f3]
vars = ["x1", "x2", "x3"]

x100 = multi_variate_newton(fs, vars, np.arange(3, dtype="float64") +1, roots=np.ones(3, dtype="float64"))
print(x100)

Error after 0 iteration 0.8558975105001576
Error after 1 iteration 0.5434872554427782
Error after 2 iteration 0.9359007498430079
Error after 3 iteration 0.5690061794192101
Error after 4 iteration 0.5356166277823504
Error after 5 iteration 0.20744843569199925
Error after 6 iteration 0.05287690803517455
Error after 7 iteration 0.0006332361575052114
Error after 8 iteration 5.388830773961401e-06
Error after 9 iteration 2.2154840111600825e-05
Error after 10 iteration 0.0022149473667646743
Error after 11 iteration 0.0
Error after 12 iteration 0.0
Error after 13 iteration 0.0
Error after 14 iteration 0.0
Error after 15 iteration 0.0
Error after 16 iteration 0.0
Error after 17 iteration 0.0
Error after 18 iteration 0.0
Error after 19 iteration 0.0
Error after 20 iteration 0.0
Error after 21 iteration 0.0
Error after 22 iteration 0.0
Error after 23 iteration 0.0
Error after 24 iteration 0.0
Error after 25 iteration 0.0
Error after 26 iteration 0.0
Error after 27 iteration 0.0
Error after 28 ite

In [3]:
#Ex2
arr = np.array([[3, 1, 1, 0],
                [1, 6, 3, -1],
                [6, 0, 9, -2],
                [1, 0, -1, -7]], dtype="float64")
b = np.ones(4, dtype="float64")
x0 = np.array([0, 1, 1, 0], dtype="float64")

x = iterate(arr, b, x0, showlen=True)
print(x)
print(np.linalg.solve(arr, b))

length of x after 0 iterations:  0.5623171296482016
length of x after 1 iterations:  0.5436427476586335
length of x after 2 iterations:  0.32284898633274967
length of x after 3 iterations:  0.4871910564375569
length of x after 4 iterations:  0.362741162129047
length of x after 5 iterations:  0.4312139362918735
length of x after 6 iterations:  0.3669717253776361
length of x after 7 iterations:  0.40157963929764967
length of x after 8 iterations:  0.3754551706387536
length of x after 9 iterations:  0.3923398647918965
length of x after 10 iterations:  0.3812080351679778
length of x after 11 iterations:  0.38876241063282185
length of x after 12 iterations:  0.3837718810842632
length of x after 13 iterations:  0.3870948360579948
length of x after 14 iterations:  0.3848641925436514
length of x after 15 iterations:  0.386340460537478
length of x after 16 iterations:  0.3853513202086695
length of x after 17 iterations:  0.38600923246822033
length of x after 18 iterations:  0.3855704238381171
l

In [4]:
#Ex3a
arr = np.array([[15, -5, 1, 1.1],
                [0, 5, 2, -1],
                [2, -1, 9, -1],
                [1, 1.1, -1, -6]], dtype="float64")
b = np.ones(4, dtype="float64")
x0 = np.array([2, 1, 1, 1], dtype="float64")

x = iterate(arr, b, x0, method="seidel", showlen=True)
print(x)
print(np.linalg.solve(arr, b))

length of x after 0 iterations:  0.3425853853095739
length of x after 1 iterations:  0.21582595589832845
length of x after 2 iterations:  0.235838816065199
length of x after 3 iterations:  0.24143190913787496
length of x after 4 iterations:  0.2422315121053915
length of x after 5 iterations:  0.24232358056664285
length of x after 6 iterations:  0.24233161156634384
length of x after 7 iterations:  0.24233226708359956
length of x after 8 iterations:  0.24233227127258744
length of x after 9 iterations:  0.24233227183216235
length of x after 10 iterations:  0.2423322703278554
length of x after 11 iterations:  0.24233227038034105
length of x after 12 iterations:  0.24233227033029536
length of x after 13 iterations:  0.24233227033715998
length of x after 14 iterations:  0.24233227033525723
length of x after 15 iterations:  0.24233227033565297
length of x after 16 iterations:  0.24233227033556456
length of x after 17 iterations:  0.2423322703355843
length of x after 18 iterations:  0.24233227

## Answers Obtained
# 1. 
    0.2082

# 2. 
    length of x after 24 iterations:  0.3857

# 3. 
    length of x after 10 iterations:  0.2423
# 4.
    2.2
# 5.
    T, F, F, F



